SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/CrossServerRPCCube.cpp

```diff
@@ -1,23 +1,50 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 
 #include "CrossServerRPCCube.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
 #include "Net/UnrealNetwork.h"
 
 ACrossServerRPCCube::ACrossServerRPCCube()
 {
 	bAlwaysRelevant = true;
 	bNetLoadOnClient = true;
 	bNetLoadOnNonAuthServer = true;
+
+	bReplicates = false;
+	SetReplicateMovement(false);
 }
 
 void ACrossServerRPCCube::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
 {
 	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
 
 	DOREPLIFETIME(ACrossServerRPCCube, ReceivedCrossServerRPCS);
+	DOREPLIFETIME(ACrossServerRPCCube, AuthEntityId);
 }
 
 void ACrossServerRPCCube::CrossServerTestRPC_Implementation(int SendingServerID)
 {
 	ReceivedCrossServerRPCS.Add(SendingServerID);
 }
+
+void ACrossServerRPCCube::TurnOnReplication()
+{
+	SetReplicates(true);
+	SetReplicateMovement(true);
+}
+
+void ACrossServerRPCCube::SetNonAuth()
+{
+	SetRole(ROLE_SimulatedProxy);
```

Comment:
My preference is just to just set these roles directly

Reply:
Sure, done